### ISSUE TITLE
2.3.3 build fails on Fedora

### ISSUE THREAD

**Issue by ignatenkobrain** [Byron/dua-cli#42]
```rust
     Running `/usr/bin/rustc --crate-name dua --edition=2018 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --emit=dep-info,link -C opt-level=3 --test -C metadata=0ead1ce69f69bace -C extra-filename=-0ead1ce69f69bace --out-dir /builddir/build/BUILD/dua-cli-2.3.3/target/release/deps -L dependency=/builddir/build/BUILD/dua-cli-2.3.3/target/release/deps --extern atty=/builddir/build/BUILD/dua-cli-2.3.3/target/release/deps/libatty-7ab650f33e255e65.rlib --extern byte_unit=/builddir/build/BUILD/dua-cli-2.3.3/target/release/deps/libbyte_unit-42624cc2036cae21.rlib --extern failure=/builddir/build/BUILD/dua-cli-2.3.3/target/release/deps/libfailure-54e08db737ce28f0.rlib --extern failure_tools=/builddir/build/BUILD/dua-cli-2.3.3/target/release/deps/libfailure_tools-ab5bfbd7d01c5737.rlib --extern filesize=/builddir/build/BUILD/dua-cli-2.3.3/target/release/deps/libfilesize-e5eca7b6ec84907b.rlib --extern itertools=/builddir/build/BUILD/dua-cli-2.3.3/target/release/deps/libitertools-a6ecbcf4956e2494.rlib --extern jwalk=/builddir/build/BUILD/dua-cli-2.3.3/target/release/deps/libjwalk-33125c89b6c104b4.rlib --extern log=/builddir/build/BUILD/dua-cli-2.3.3/target/release/deps/liblog-c2673a3a8764d8d2.rlib --extern num_cpus=/builddir/build/BUILD/dua-cli-2.3.3/target/release/deps/libnum_cpus-540330b3b6fd3719.rlib --extern open=/builddir/build/BUILD/dua-cli-2.3.3/target/release/deps/libopen-99a9b03245640124.rlib --extern petgraph=/builddir/build/BUILD/dua-cli-2.3.3/target/release/deps/libpetgraph-2b4870a46b520600.rlib --extern pretty_assertions=/builddir/build/BUILD/dua-cli-2.3.3/target/release/deps/libpretty_assertions-e03e860cdfe699c4.rlib --extern structopt=/builddir/build/BUILD/dua-cli-2.3.3/target/release/deps/libstructopt-953ea39fa9ebfd19.rlib --extern termion=/builddir/build/BUILD/dua-cli-2.3.3/target/release/deps/libtermion-8fdc6d3b98684c34.rlib --extern tui=/builddir/build/BUILD/dua-cli-2.3.3/target/release/deps/libtui-0c3680176c266e4e.rlib --extern tui_react=/builddir/build/BUILD/dua-cli-2.3.3/target/release/deps/libtui_react-6211bea35766d921.rlib --extern unicode_segmentation=/builddir/build/BUILD/dua-cli-2.3.3/target/release/deps/libunicode_segmentation-9fede8a1cd057827.rlib -Copt-level=3 -Cdebuginfo=2 -Clink-arg=-Wl,-z,relro,-z,now -Ccodegen-units=1 --cap-lints=warn -L native=/builddir/build/BUILD/dua-cli-2.3.3/target/release/build/backtrace-sys-65450aba3fc6d909/out`
error[E0308]: mismatched types
  --> src/interactive/app_test/journeys_readonly.rs:50:28
   |
50 |         app.process_events(&mut terminal, b"s".keys())?;
   |                            ^^^^^^^^^^^^^
   |                            |
   |                            expected struct `tui_react::terminal::Terminal`, found mutable reference
   |                            help: consider removing the borrow: `terminal`
   |
   = note:         expected struct `tui_react::terminal::Terminal<_>`
           found mutable reference `&mut tui_react::terminal::Terminal<tui::backend::test::TestBackend>`
error[E0308]: mismatched types
  --> src/interactive/app_test/journeys_readonly.rs:62:28
   |
62 |         app.process_events(&mut terminal, b"s".keys())?;
   |                            ^^^^^^^^^^^^^
   |                            |
   |                            expected struct `tui_react::terminal::Terminal`, found mutable reference
   |                            help: consider removing the borrow: `terminal`
   |
   = note:         expected struct `tui_react::terminal::Terminal<_>`
           found mutable reference `&mut tui_react::terminal::Terminal<tui::backend::test::TestBackend>`
error[E0308]: mismatched types
  --> src/interactive/app_test/journeys_readonly.rs:78:28
   |
78 |         app.process_events(&mut terminal, b"j".keys())?;
   |                            ^^^^^^^^^^^^^
   |                            |
   |                            expected struct `tui_react::terminal::Terminal`, found mutable reference
   |                            help: consider removing the borrow: `terminal`
   |
   = note:         expected struct `tui_react::terminal::Terminal<_>`
           found mutable reference `&mut tui_react::terminal::Terminal<tui::backend::test::TestBackend>`
error[E0308]: mismatched types
  --> src/interactive/app_test/journeys_readonly.rs:85:28
   |
85 |         app.process_events(&mut terminal, b"j".keys())?;
   |                            ^^^^^^^^^^^^^
   |                            |
   |                            expected struct `tui_react::terminal::Terminal`, found mutable reference
   |                            help: consider removing the borrow: `terminal`
   |
   = note:         expected struct `tui_react::terminal::Terminal<_>`
           found mutable reference `&mut tui_react::terminal::Terminal<tui::backend::test::TestBackend>`
error[E0308]: mismatched types
  --> src/interactive/app_test/journeys_readonly.rs:92:28
   |
92 |         app.process_events(&mut terminal, b"k".keys())?;
   |                            ^^^^^^^^^^^^^
   |                            |
   |                            expected struct `tui_react::terminal::Terminal`, found mutable reference
   |                            help: consider removing the borrow: `terminal`
   |
   = note:         expected struct `tui_react::terminal::Terminal<_>`
           found mutable reference `&mut tui_react::terminal::Terminal<tui::backend::test::TestBackend>`
error[E0308]: mismatched types
  --> src/interactive/app_test/journeys_readonly.rs:99:28
   |
99 |         app.process_events(&mut terminal, b"k".keys())?;
   |                            ^^^^^^^^^^^^^
   |                            |
   |                            expected struct `tui_react::terminal::Terminal`, found mutable reference
   |                            help: consider removing the borrow: `terminal`
   |
   = note:         expected struct `tui_react::terminal::Terminal<_>`
           found mutable reference `&mut tui_react::terminal::Terminal<tui::backend::test::TestBackend>`
error[E0308]: mismatched types
   --> src/interactive/app_test/journeys_readonly.rs:106:28
    |
106 |         app.process_events(&mut terminal, b"o".keys())?;
    |                            ^^^^^^^^^^^^^
    |                            |
    |                            expected struct `tui_react::terminal::Terminal`, found mutable reference
    |                            help: consider removing the borrow: `terminal`
    |
    = note:         expected struct `tui_react::terminal::Terminal<_>`
            found mutable reference `&mut tui_react::terminal::Terminal<tui::backend::test::TestBackend>`
error[E0308]: mismatched types
   --> src/interactive/app_test/journeys_readonly.rs:120:32
    |
120 |             app.process_events(&mut terminal, b"u".keys())?;
    |                                ^^^^^^^^^^^^^
    |                                |
    |                                expected struct `tui_react::terminal::Terminal`, found mutable reference
    |                                help: consider removing the borrow: `terminal`
    |
    = note:         expected struct `tui_react::terminal::Terminal<_>`
            found mutable reference `&mut tui_react::terminal::Terminal<tui::backend::test::TestBackend>`
error[E0308]: mismatched types
   --> src/interactive/app_test/journeys_readonly.rs:135:28
    |
135 |         app.process_events(&mut terminal, b"ju".keys())?;
    |                            ^^^^^^^^^^^^^
    |                            |
    |                            expected struct `tui_react::terminal::Terminal`, found mutable reference
    |                            help: consider removing the borrow: `terminal`
    |
    = note:         expected struct `tui_react::terminal::Terminal<_>`
            found mutable reference `&mut tui_react::terminal::Terminal<tui::backend::test::TestBackend>`
error[E0308]: mismatched types
   --> src/interactive/app_test/journeys_readonly.rs:152:28
    |
152 |         app.process_events(&mut terminal, b"k".keys())?;
    |                            ^^^^^^^^^^^^^
    |                            |
    |                            expected struct `tui_react::terminal::Terminal`, found mutable reference
    |                            help: consider removing the borrow: `terminal`
    |
    = note:         expected struct `tui_react::terminal::Terminal<_>`
            found mutable reference `&mut tui_react::terminal::Terminal<tui::backend::test::TestBackend>`
error[E0308]: mismatched types
   --> src/interactive/app_test/journeys_readonly.rs:154:28
    |
154 |         app.process_events(&mut terminal, b"d".keys())?;
    |                            ^^^^^^^^^^^^^
    |                            |
    |                            expected struct `tui_react::terminal::Terminal`, found mutable reference
    |                            help: consider removing the borrow: `terminal`
    |
    = note:         expected struct `tui_react::terminal::Terminal<_>`
            found mutable reference `&mut tui_react::terminal::Terminal<tui::backend::test::TestBackend>`
error[E0308]: mismatched types
   --> src/interactive/app_test/journeys_readonly.rs:176:32
    |
176 |             app.process_events(&mut terminal, b"d".keys())?;
    |                                ^^^^^^^^^^^^^
    |                                |
    |                                expected struct `tui_react::terminal::Terminal`, found mutable reference
    |                                help: consider removing the borrow: `terminal`
    |
    = note:         expected struct `tui_react::terminal::Terminal<_>`
            found mutable reference `&mut tui_react::terminal::Terminal<tui::backend::test::TestBackend>`
error[E0308]: mismatched types
   --> src/interactive/app_test/journeys_readonly.rs:193:32
    |
193 |             app.process_events(&mut terminal, b"d".keys())?;
    |                                ^^^^^^^^^^^^^
    |                                |
    |                                expected struct `tui_react::terminal::Terminal`, found mutable reference
    |                                help: consider removing the borrow: `terminal`
    |
    = note:         expected struct `tui_react::terminal::Terminal<_>`
            found mutable reference `&mut tui_react::terminal::Terminal<tui::backend::test::TestBackend>`
error[E0308]: mismatched types
   --> src/interactive/app_test/journeys_readonly.rs:210:32
    |
210 |             app.process_events(&mut terminal, b"k ".keys())?;
    |                                ^^^^^^^^^^^^^
    |                                |
    |                                expected struct `tui_react::terminal::Terminal`, found mutable reference
    |                                help: consider removing the borrow: `terminal`
    |
    = note:         expected struct `tui_react::terminal::Terminal<_>`
            found mutable reference `&mut tui_react::terminal::Terminal<tui::backend::test::TestBackend>`
error[E0308]: mismatched types
   --> src/interactive/app_test/journeys_readonly.rs:229:28
    |
229 |         app.process_events(&mut terminal, b" j ".keys())?;
    |                            ^^^^^^^^^^^^^
    |                            |
    |                            expected struct `tui_react::terminal::Terminal`, found mutable reference
    |                            help: consider removing the borrow: `terminal`
    |
    = note:         expected struct `tui_react::terminal::Terminal<_>`
            found mutable reference `&mut tui_react::terminal::Terminal<tui::backend::test::TestBackend>`
error[E0308]: mismatched types
   --> src/interactive/app_test/journeys_readonly.rs:243:28
    |
243 |         app.process_events(&mut terminal, b"\t".keys())?;
    |                            ^^^^^^^^^^^^^
    |                            |
    |                            expected struct `tui_react::terminal::Terminal`, found mutable reference
    |                            help: consider removing the borrow: `terminal`
    |
    = note:         expected struct `tui_react::terminal::Terminal<_>`
            found mutable reference `&mut tui_react::terminal::Terminal<tui::backend::test::TestBackend>`
error[E0308]: mismatched types
  --> src/interactive/app_test/journeys_with_writes.rs:15:24
   |
15 |     app.process_events(&mut terminal, b"doddd".keys())?;
   |                        ^^^^^^^^^^^^^
   |                        |
   |                        expected struct `tui_react::terminal::Terminal`, found mutable reference
   |                        help: consider removing the borrow: `terminal`
   |
   = note:         expected struct `tui_react::terminal::Terminal<_>`
           found mutable reference `&mut tui_react::terminal::Terminal<tui::backend::test::TestBackend>`
error[E0308]: mismatched types
  --> src/interactive/app_test/journeys_with_writes.rs:31:9
   |
31 |         &mut terminal,
   |         ^^^^^^^^^^^^^
   |         |
   |         expected struct `tui_react::terminal::Terminal`, found mutable reference
   |         help: consider removing the borrow: `terminal`
   |
   = note:         expected struct `tui_react::terminal::Terminal<_>`
           found mutable reference `&mut tui_react::terminal::Terminal<tui::backend::test::TestBackend>`
```

**Comment by Byron** [comment]
Sorry for the hassle.

The course of events: I made a change, ran `make test`, which fails locally due to some white-space/line ending issue on MacOS but no Linux. I failed to run journey tests manually and thus didn't discover the issue above. Nor did I run cargo check from the IDE, which could also have discovered the issue.

When travis sent an email after the release was done, I took a quick look and saw that one build failed as it tried to run 32bit MacOS. This, I though, could wait for a moment, and I ignored/failed to look at the other failure which is the one above.

Version 2.3.3. was yanked as CI didn't pass.
